Simulation assertion failed findLimitingSignal

### What happened?

When running a simulation a core error occurred:

```
[14:15:46,420] [ERROR] [ScheduleMetadataExtractor] signal XXXXXXXXXXXXXXXXXXXXXXXXXX does not react to zone occupation
...
	at fr.sncf.osrd.standalone_sim.ScheduleMetadataExtractor.findLimitingSignal(ScheduleMetadataExtractor.kt:387)
	at fr.sncf.osrd.standalone_sim.ScheduleMetadataExtractor.routingRequirements_ux9p5c4$findRouteSetDeadline(ScheduleMetadataExtractor.kt:288)
	at fr.sncf.osrd.standalone_sim.ScheduleMetadataExtractor.routingRequirements-ux9p5c4(ScheduleMetadataExtractor.kt:309)
	at fr.sncf.osrd.standalone_sim.ScheduleMetadataExtractor.run(ScheduleMetadataExtractor.kt:144)
	at fr.sncf.osrd.standalone_sim.StandaloneSim.run(StandaloneSim.java:94)
	at fr.sncf.osrd.standalone_sim.StandaloneSim.runFromRJS(StandaloneSim.java:145)
	at fr.sncf.osrd.api.StandaloneSimulationEndpoint.act(StandaloneSimulationEndpoint.java:69)
```

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to "Operational Studies"
2. Create a Project / Study / Scenario
3. Select any rolling stock and path
4. Create a train

You can now see the stack trace of osrd-core.

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local & Dev


### OSRD version (top right corner `Account` button > `Informations`)

9c781be3fb88eebb3a9f535c99338f3a9dd5eef4

--- a/core/src/main/java/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractor.kt
+++ b/core/src/main/java/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractor.kt
@@ -254,7 +254,7 @@ private fun routingRequirements(
         curOffset += blockLength
     }
 
-    fun findRouteSetDeadline(routeIndex: Int): Double {
+    fun findRouteSetDeadline(routeIndex: Int): Double? {
         if (routeIndex == 0)
             // TODO: this isn't quite true when the path starts with a stop
             return 0.0
@@ -286,6 +286,7 @@ private fun routingRequirements(
         // find the first non-open signal on the path
         // iterate backwards on blocks from blockIndex to 0, and on signals
         val limitingSignalSpec = findLimitingSignal(blockInfra, simulatedSignalStates, blockPath, routeStartBlockIndex)
+            ?: return null
         val limitingBlock = blockPath[limitingSignalSpec.blockIndex]
         val signal = blockInfra.getBlockSignals(limitingBlock)[limitingSignalSpec.signalIndex]
         val signalOffset = blockInfra.getSignalsPositions(limitingBlock)[limitingSignalSpec.signalIndex]
@@ -306,7 +307,7 @@ private fun routingRequirements(
     for (routeIndex in 0 until routePath.size) {
         // start out by figuring out when the route needs to be set
         // when the route is set, signaling can allow the train to proceed
-        val routeSetDeadline = findRouteSetDeadline(routeIndex)
+        val routeSetDeadline = findRouteSetDeadline(routeIndex)?: continue
 
         // find the release time of the last zone of each release group
         val route = routePath[routeIndex]
@@ -367,7 +368,7 @@ private fun findLimitingSignal(
     simulatedSignalStates: IdxMap<LogicalSignalId, SigState>,
     blockPath: StaticIdxList<Block>,
     routeStartBlockIndex: Int
-): LimitingSignal {
+): LimitingSignal? {
     var lastSignalBlockIndex = -1
     var lastSignalIndex = -1
     for (curBlockIndex in (0 until routeStartBlockIndex).reversed()) {
@@ -384,7 +385,9 @@ private fun findLimitingSignal(
             lastSignalIndex = curSignalIndex
         }
     }
-    assert(lastSignalBlockIndex != -1 && lastSignalIndex != -1)
+    // Limiting signal not found
+    if (lastSignalBlockIndex == -1 || lastSignalIndex == -1)
+        return null
     return LimitingSignal(lastSignalBlockIndex, lastSignalIndex)
 }
 

